db/Pos/DB/GState/Common.hs

```diff
@@ -143,8 +164,21 @@ getTipMaybe = gsGetBi tipKey
 getBotMaybe :: MonadDBRead m => m (Maybe HeaderHash)
 getBotMaybe = gsGetBi botKey
 
+getMaxSeenDifficultyMaybe :: MonadDBRead m => m (Maybe ChainDifficulty)
+getMaxSeenDifficultyMaybe = gsGetBi maxSeenDifficultyKey
+
 putTip :: MonadDB m => HeaderHash -> m ()
 putTip = gsPutBi tipKey
 
 putBot :: MonadDB m => HeaderHash -> m ()
 putBot = gsPutBi botKey
+
+putMaxSeenDifficulty :: MonadDB m => ChainDifficulty -> m ()
+putMaxSeenDifficulty = gsPutBi maxSeenDifficultyKey
+
+-- | Put difficulty only if it's bigger than existing one.
+updateMaxSeenDifficulty :: MonadDB m => ChainDifficulty -> m ()
```

Comment:
I think this function should be removed. It's not used and if it was used it would most likely be a bug. That's because we should update GState only in batches. Instead of using `putMaxSeenDifficulty` you can return `m (Maybe CommonOp`) (or `m [CommonOp]`).